docs/admin/disruptions.md

```diff
@@ -0,0 +1,87 @@
+---
+assignees:
+- mml
+
+---
+This guide is for anyone wishing to specify safety constraints on pods or anyone
+wishing to write software (typically automation software) that respects those
+constraints.
+
+* TOC
+{:toc}
+
+## Rationale
+
+Various cluster management operations may voluntarily evict pods.  "Voluntary"
+means an eviction can be safely delayed for a reasonable period of time. The
+principal examples today are draining a node for maintenance or upgrade
+(`kubectl drain`), and cluster autoscaling down. In the future the
+[rescheduler](https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/rescheduling.md)
+may also perform voluntary evictions.  By contrast, something like evicting pods
+because a node has become unreachable or reports NotReady, is not "voluntary."
+
+For voluntary evictions, it can be useful for applications to be able to limit
+the number of pods that are down, or the rate with which pods are evicted.  For
+example, a quorum-based application would like to ensure that the number of
+replicas running is never brought below the number needed for a quorum, even
+temporarily. Or a web front end might want to ensure that the number of replicas
+serving load never falls below a certain percentage of the total, even briefly.
+`PodDisruptionBudget` is an API object that specifies the minimum number or
+percentage of replicas of a collection that must be up at a time, and/or the
+maximum eviction rate across the collection. Components that wish to eviction a
+pod subject to disruption budget use the `/eviction` subresource; unlike a
+regular pod deletion, this operation may be rejected by the API server if the
+eviction would cause a disruption budget to be violated.
+
+## Specifying a PodDisruptionBudget
+
+A `PodDisruptionBudget` has two components: a selector to specify the set of
```

Comment:
The "i.e." is not how it's implemented.  This may be a simple off-by-one error, or it may be a place where we didn't think carefully enough about the spec.

As it stands, if you say "100%", we will allow evictions if all the pods are ready.  If we want "100%" to mean "never permit a 'voluntary' eviction", I will clarify the docs, and changing the code and tests will be easy.

Or maybe the right thing is to open an issue to discuss and track this, and for now document the way it works?